render/download.go

```diff
@@ -0,0 +1,62 @@
+package render
+
+import (
+	"fmt"
+	"io"
+	"mime"
+	"net/http"
+	"path/filepath"
+	"strconv"
+)
+
+type downloadRenderer struct {
+	data     []byte
+	filename string
+	writer   http.ResponseWriter
+}
+
+func (r downloadRenderer) ContentType() string {
+	ext := filepath.Ext(r.filename)
+	t := mime.TypeByExtension(ext)
+	if t == "" {
+		t = "application/octet-stream"
+	}
+
+	return t
+}
+
+func (r downloadRenderer) Render(w io.Writer, d Data) error {
+	r.writer.Header().Add("Content-Disposition", fmt.Sprintf("attachment; filename=%s", r.filename))
```

Comment:
Cast w if it’s an http writer,or better yet it’s a writer that implements an interface that has Header() then write the headers.

Reply:
Yes, sorry, you're right. That's what happens when you do PR reviews before coffee. :)